clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java

```diff
@@ -379,7 +379,9 @@ private void onErrorResponse(final R response, final long currentTimeMs) {
             case GROUP_MAX_SIZE_REACHED:
             case UNSUPPORTED_ASSIGNOR:
             case UNSUPPORTED_VERSION:
-                logger.error("{} failed due to {}: {}", heartbeatRequestName(), error, errorMessage);
+                logger.error("{} failed due to {}: {}. The cluster doesn't yet support the new consumer group protocol." +
+                        " Set group.protocol=classic to revert to the classic protocol until the cluster is upgraded.",
+                        heartbeatRequestName(), error, errorMessage);
                 handleFatalFailure(error.exception(errorMessage));
```

Comment:
I think that we should also use the custom error message for `UNSUPPORTED_VERSION` in order to ensure that the end user gets it.